includes/class-convertkit-broadcasts-importer.php

```diff
@@ -73,143 +90,201 @@ public function refresh( $broadcasts ) {
 			}
+		} else {
+			$this->maybe_log( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Skipping thumbnail.' );
+		}
 
-			// If the Import Thumbnail setting is enabled, and the Broadcast has an image, save it to the Media Library and link it to the Post.
-			if ( $this->broadcasts_settings->import_thumbnail() ) {
-				$result = $this->add_broadcast_image_to_post( $broadcast, $post_id );
+		// Transition the Post to the defined Post Status in the settings, now that the image has been added to it.
+		$post_id = wp_update_post(
+			array(
+				'ID'          => $post_id,
+				'post_status' => $post_status,
+			),
+			true
+		);
 
-				if ( is_wp_error( $result ) && $settings->debug_enabled() ) {
-					$log->add( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Error on add_broadcast_image_to_post(): ' . $result->get_error_message() );
-				}
-			} elseif ( $settings->debug_enabled() ) {
-				$log->add( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Skipping thumbnail.' );
-			}
+		// Maybe log if an error occured updating the Post to the publish status.
+		if ( is_wp_error( $post_id ) ) {
+			$this->maybe_log( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Error on wp_update_post() when transitioning post status from draft to publish: ' . $post_id->get_error_message() );
+			return $post_id;
+		}
 
-			// Transition the Post to the defined Post Status in the settings, now that the image has been added to it.
-			$post_id = wp_update_post(
-				array(
-					'ID'          => $post_id,
-					'post_status' => $this->broadcasts_settings->post_status(),
-				),
-				true
-			);
+		// Import successful.
+		$this->maybe_log( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Added as Post ID #' . $post_id );
+		return $post_id;
 
-			// Maybe log if an error occured updating the Post to the publish status.
-			if ( is_wp_error( $post_id ) ) {
-				if ( $settings->debug_enabled() ) {
-					$log->add( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Error on wp_update_post() when transitioning post status from draft to publish: ' . $post_id->get_error_message() );
-				}
-			}
-			if ( $settings->debug_enabled() ) {
-				$log->add( 'ConvertKit_Broadcasts_Importer::refresh(): Broadcast #' . $broadcast_id . '. Added as Post ID #' . $post_id );
-			}
+	}
+
+	/**
+	 * Logs the given message if debug logging is enabled
+	 * in the Plugin's settings.
+	 *
+	 * @since   2.6.4
+	 *
+	 * @param   string $message    Log message.
+	 */
+	private function maybe_log( $message ) {
+
+		// Don't log if debugging is not enabled.
+		if ( ! $this->settings->debug_enabled() ) {
+			return;
 		}
 
+		// Initialize logging class, if not yet initialized.
+		if ( ! $this->log ) {
```

Comment:
I would expect this check and initialization in the constructor.

Reply:
Agreed, however PHPStan throws an error as it believes doing this in the constructor would mean key WordPress files used for logging aren't loaded:
![Screenshot 2024-11-01 at 09 28 35](https://github.com/user-attachments/assets/2ddcf8ad-537b-4b2d-a292-e0d327631ec6)

Not sure why this is the case, so will review in a future PR.